src/cache-utils.ts

```diff
@@ -47,11 +51,24 @@ export const getCommandOutput = async (toolCommand: string) => {
   return stdout.trim();
 };
 
-const getPackageManagerVersion = async (
+export const getPackageManagerWorkingDir = (): string | null => {
+  const cache = core.getInput('cache');
+  if (cache !== 'yarn') {
+    return null;
+  }
+
+  const cacheDependencyPath = core.getInput('cache-dependency-path');
+  return cacheDependencyPath ? path.dirname(cacheDependencyPath) : null;
```

Comment:
we can change return default return argument to undefined.

Reply:
the function returning undefined suspected to be void, and `undefined` means "unassigned" while i want to assign and return meaningful value "no path"